tests/console/sysstat.pm

```diff
@@ -72,14 +75,17 @@ sub run {
     validate_script_output "mpstat",     sub { /CPU    %usr   %nice    %sys %iowait    %irq   %soft  %steal  %guest  %gnice   %idle/ };
     validate_script_output "sar -u",     sub { /CPU     %user     %nice   %system   %iowait    %steal     %idle/ };
     validate_script_output "sar -n DEV", sub { /IFACE   rxpck\/s   txpck\/s    rxkB\/s    txkB\/s   rxcmp\/s   txcmp\/s  rxmcst\/s   %ifutil/ };
-    if (is_sle('<=15-SP3') || is_leap('<=15.3')) {
-        validate_script_output "sar -b", sub { /tps      rtps      wtps   bread\/s   bwrtn\/s/ };
-    } else {
+    #from version 12.1.2 iostat supports discard I/O statistics.
+    if (version->parse(script_output('rpm --qf "%{VERSION}\n" -q sysstat')) >= version->parse('12.1.2')) {
         validate_script_output "sar -b", sub { /tps      rtps      wtps      dtps   bread\/s   bwrtn\/s   bdscd\/s/ };
+    } else {
+        validate_script_output "sar -b", sub { /tps      rtps      wtps   bread\/s   bwrtn\/s/ };
     }
     validate_script_output "sar -B", sub { /pgpgin\/s pgpgout\/s   fault\/s  majflt\/s  pgfree\/s pgscank\/s pgscand\/s pgsteal\/s    %vmeff/ };
     validate_script_output "sar -H", sub { /kbhugfree kbhugused  %hugused/ };
     validate_script_output "sar -S", sub { /kbswpfree kbswpused  %swpused  kbswpcad   %swpcad/ };
+
+    assert_script_run 'unset S_COLORS';
```

Comment:
to be pedantic, this isn't the same as before...
and we try to disable the colors but if it was `unset` before the colors weren't enabled. I guess if you really wanna be that untouchy, then you need to safe the old value. Or set it only for your `sar` calls.
But from my perspective, I would not care.

Reply:
> to be pedantic, this isn't the same as before...
> and we try to disable the colors but if it was `unset` before the colors weren't enabled. I guess if you really wanna be that untouchy, then you need to safe the old value. Or set it only for your `sar` calls.

It should not be set by default.

```
# ltrace -e getenv sar -b
sar->getenv("S_COLORS")                                                                                               = nil
sar->getenv("S_COLORS_SGR")                                                                                           = nil
sar->getenv("S_TIME_DEF_TIME")                                                                                        = nil
sar->getenv("S_TIME_FORMAT")                                                                                          = nil
Linux 5.7.9-1-default (kepler)  09/02/20        _x86_64_        (8 CPU)

11:10:55     LINUX RESTART      (8 CPU)

12:19:29          tps      rtps      wtps      dtps   bread/s   bwrtn/s   bdscd/s
14:09:56         0.00      0.00      0.00      0.00      0.00      0.00      0.00
14:10:01       115.00      0.40    114.60      0.00     19.20   8734.40      0.00
14:10:06        63.40      0.40     63.00      0.00      8.00   1954.40      0.00
14:10:11        95.20      0.20     95.00      0.00      1.60   2764.80      0.00
14:10:16       103.60      0.80    102.80      0.00     25.60   5019.20      0.00
Average:         0.00      0.00      0.00      0.00      0.00      0.00      0.00
+++ exited (status 0) +++
```
> But from my perspective, I would not care and just leave it with `never`.

Reply:
`ltrace -e getenv` nice, thx for that hint!

My fault, I didn't read the manpage properly. There it say `auto` is the default. So it is very likely that S_COLORS is just not set.

Nothing to add from my side :)

Reply:
Your concerns are valid thus it is always better double-checked it. It was a nice exercise. :) Thanks!